### PR TITLE
Gaj/feature/traveler test

### LINKED ISSUE #10: Traveler Class & Testing
##About:
> Describe this feature’s purpose here.

This builds out the Traveler Class. 

### Story: 
As a dev, I want to have a Traveler class, so that I can pass in an object as an argument and return data pertaining to a specific Traveler. 

#### Concerns: 
- [x] does the constructor contain properties to hold a Travelers data
- [x] should find all of traveler's trips
- [x] should filter if their trips are approved or pending
- [x] should calculate total amount of money spent on travels, total
- [x] should find past trips
- [x] should find current trips
- [x] should find future trips

### PR DESCRIPTION
**Pull request by gaj23**
- [x] Feature
- [ ] Refactor
- [ ] Bug Fix 
- [ ] Clean Up

This PR addressed the Traveler class buildout as well as its associated testing. 

To confirm functionality, as, well, a random Traveler was selected and the greeting and "stats" section were updated accordingly. 

Small additional changes were made to the appearance to take into account the need for a home button when navigating away from the home page. 

<details>
<summary>Previous Branch</summary>
<br>
Screen Shot 2021-01-16 at 2 11 09 PM
Screen Shot 2021-01-16 at 3 04 50 PM
</details>
<details>
<summary>Current Branch</summary>
<br>
Screen Shot 2021-01-17 at 7 06 41 PM
Screen Shot 2021-01-17 at 7 06 31 PM
</details>

This PR closes #10 and #12 of issues.

When building out Traveler, significant effort was put into how to find current, planned, and past trips from the data. After researching options and trying various patterns, I decided to implement [Luxon](https://moment.github.io/luxon/docs/manual/install.html) rather than moment.js due to the approachability of the code as well as the size of the dependency. Once Luxon was implemented, I found it incredibly easy to compare dates. This function, however, ended up being fairly hefty and could use attention with refactoring. 

Please see src/Travelers.js and test/travelers-test.js to see those changes. 

For the beginning of the domUpdate please see src/domUpdates.js.

Smaller changes were made inside of src/css 
- for logo changes see src/css/base.scss line 21

Additionally, see those reflected structural changes in src/index.html line 15 - 19.

First please download the dependency: `npm install --save luxon` 

Then, please fun npm test test/traveler-test.js to confirm all tests pass.

Dom manipulation so that the appearance of a specific travelers data occurs in a semi-pleasing manner. Continued building of classes, as well. 

#### Checklist:
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My code follows the style guidelines of this project
- [x] I have used tests to prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my change

### CODE REVIEW

File: src/scripts.js [hunk 1]
```diff
@@ -22,22 +35,17 @@ window.onload = Promise.all([apiCalls.getTravelersData(), apiCalls.getTripsData(
...
 
+function instantiation() {
+  traveler = new Traveler(travelers[7]);
+  domUpdates.greetTraveler(traveler);
+  domUpdates.updateTravelerStats(traveler, trips, destinations);
+}
 
 function toggleDarkMode() {
-  document.querySelector('body').classList.toggle('dark-mode-on');
   darkLightModeButton.classList.toggle('dark-mode-on');
```
Comment (srslie) [inline src/scripts.js:48]: Really nice refactor to clean this up!

File: test/traveler-test.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    expect(traveler2.findTotalYearExpense(2019, tripsData, destinationsData)).to.equal(5819);
+    expect(traveler2.findTotalYearExpense(2021, tripsData, destinationsData)).to.equal(6094);
+  })
+
+  it('Should separate pending and approved trips', () => {
+    expect(traveler.separateApprovedTrips(tripsData).approved.length).to.equal(4);
+    expect(traveler.separateApprovedTrips(tripsData).pending.length).to.equal(0);
+    expect(traveler3.separateApprovedTrips(tripsData).pending.length).to.equal(1);
+  })
+
```
Comment (srslie) [inline test/traveler-test.js:45]: Nice comprehensive test with a small test file.

**Review by srslie — COMMENTED** [review]
Nice job on getting those functions written, especially with the date logic. Also loved how you're cleaning up the code with those domUpdate refactors and setting clear plans in your comments.

File: src/Traveler.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+  findTotalExpense(tripsData, destinationsData) {
+    return this.findExpenses(tripsData, destinationsData);
+  }
+
+  findTotalYearExpense(year, tripsData, destinationsData) {
+    return this.filterMyTrips(tripsData).reduce((acc, trip) => {
+      if (parseInt(trip.date.split('/')[0]) === year) {
+        destinationsData.find(destination => {
+          if (trip.destinationID === destination.id) {
```
[the remark this answers is not in the document]
Comment (gaj23) [reply src/Traveler.js:24]: Thank you for the suggestion!

File: src/Traveler.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+            acc += (subtotal * .1) + subtotal;
+          }
+        })
+      }
+      return acc;
+    }, 0)
+  }
+
+  //Would like to refactor findTotalYearExpense to encorpreate findExpenses (and cut down redundancy), but problem of 'trip' needed to be defined.
+  findExpenses(tripsData, destinationsData) {
```
[the remark this answers is not in the document]
Comment (gaj23) [reply src/Traveler.js:35]: Yes, it's trying to find expenses for all trips ever, approved or pending, past, present or future.

File: src/Traveler.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+} from 'luxon';
+
+class Traveler {
+  constructor(traveler) {
+    this.id = traveler.id;
+    this.name = traveler.name;
+    this.travelerType = traveler.travelerType;
+  }
+
+  filterMyTrips(tripsData) {
```
[the remark this answers is not in the document]
Comment (gaj23) [reply src/Traveler.js:12]: I'll put that into consideration.

File: src/Traveler.js [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+      }
+      if (endDate < currentDate) {
+        acc.past.push(trip);
+      } else if (startDate > currentDate) {
+        acc.future.push(trip);
+      } else if (startDate < currentDate && endDate > currentDate) {
+        acc.present.push(trip);
+      }
+      return acc;
+    }, {})
```
[the remark this answers is not in the document]
Comment (gaj23) [reply src/Traveler.js:85]: Thank you! That's what I thought too.

File: src/Traveler.js [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+      } else {
+        acc.pending.push(trip);
+      }
+      return acc;
+    }, {})
+  }
+
+  separateTripTimings(tripsData) {
+    return this.filterMyTrips(tripsData).reduce((acc, trip) => {
+      const currentDate = DateTime.local();
```
[the remark this answers is not in the document]
Comment (gaj23) [reply src/Traveler.js:64]: When working with new Data, I really struggled with conceptualization more than anything of what it was doing, what it was expecting, and what it was out putting. I found DateTime to be easier for my brain to digest and logic how to use.

File: src/css/_buttons.scss [hunk 8]
```diff
@@ -1,14 +1,12 @@
 @import 'mixins', 'variables';
```
[the remark this answers is not in the document]
Comment (gaj23) [reply src/css/_buttons.scss:1]: Not yet!

File: src/scripts.js [hunk 9]
```diff
@@ -22,22 +35,17 @@ window.onload = Promise.all([apiCalls.getTravelersData(), apiCalls.getTripsData(
     data[0].travelers.forEach(traveler => travelers.push(traveler))
     data[1].trips.forEach(trip => trips.push(trip))
     data[2].destinations.forEach(destination => destinations.push(destination))
+    instantiation();
   })
 
+function instantiation() {
```
[the remark this answers is not in the document]
Comment (gaj23) [reply src/scripts.js:41]: Small, but easy fix!

**Pull request merged**